server/src/auth/strategy/local.strategy.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy } from 'passport-local';
+import { AuthService } from '../auth.service';
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private readonly authService: AuthService) {
+    super();
+  }
+
+  async validate(username: string, password: string): Promise<any> {
```

Comment:
We actually do have a type for users. It's either Admin or User. Be careful using "any" since it will subvert type-checking. Also, we don't actually use passwords for authentication. We use tokens. AuthService should have something on that.